### PR TITLE
DO NOT  MERGE- Integration:  Refresh config fields

### PR DESCRIPTION
**Pull request by jimmyxian**
Depends on: https://github.com/samalba/dockerclient/pull/116

Fix: https://github.com/docker/swarm/issues/908  https://github.com/docker/swarm/issues/912
Signed-off-by: Xian Chaobo <xianchaobo@huawei.com>

### CODE REVIEW

File: test/integration/api/run.bats [hunk 1]
```diff
@@ -27,8 +27,43 @@ function teardown() {
...
+			 --device=/dev/loop0:/dev/loop0 \
+			 --ipc=host \
+			 --pid=host \
+			 busybox sleep 1000
 
 	# verify, container is running
 	[ -n $(docker_swarm ps -q --filter=name=test_container --filter=status=running) ]
+
+	# label
+	docker_swarm inspect -f '{{.Config.Labels}}' test_container | grep -q "com.example.version:1.0"
```
Comment (chanwit) [inline test/integration/api/run.bats]: It is possible to speed things up by
```
INSPECT=$(docker_swarm inspect test_container)
echo $INSPECT | grep -q ...
```
so you can just call `inspect` once.

Comment (jimmyxian) [inline test/integration/api/run.bats]: Yeah, I think so.

**Comment by vieux** [comment]
LGTM

**Comment by vieux** [comment]
@jimmyxian I merged this one manually with the right git hash. Thanks!

**Comment by jimmyxian** [comment]
@vieux OK,thanks. Do I need open a new PR to add the integration test.

**Pull request closed without merging**